test/builder/eventlistener.go

```diff
@@ -106,3 +109,47 @@ func Trigger(tbName, ttName, apiVersion string, ops ...EventListenerTriggerOp) v
 
 	return t
 }
+
+// EventListenerTriggerValidate adds a TrjggerValidate to the Triger in EventListenerSpec Triggers.
+func EventListenerTriggerValidate(ops ...EventListenerTriggerValidateOp) EventListenerTriggerOp {
+	return func(trigger *v1alpha1.EventListenerTrigger) {
+		validate := &v1alpha1.TriggerValidate{}
+		for _, op := range ops {
+			op(validate)
+		}
+		trigger.TriggerValidate = validate
```

Comment:
nit: the `trigger.TriggerValidate` already exists, so this code could be condensed a little with:
```
return func(trigger *v1alpha1.EventListenerTrigger) {
  for _, op := range ops {
    op(&trigger.TriggerValidate)
  }
}
```

Reply:
Oh, ok. I didn't realize that `TriggerValidate` was a pointer; all of the other fields are not pointers. My mistake 😁 
Out of curiosity, why is `TriggerValidate` a pointer?

Reply:
Because it is optional. We check whether it's there or not. If it's there, we do the validation.